File: app/src/main/java/com/mdgd/pokemon/ui/splash/SplashContract.java
package com.mdgd.pokemon.ui.splash;

import com.mdgd.mvi.fragments.FragmentContract;

public class SplashContract {

    public interface ViewModel extends FragmentContract.ViewModel<View> {
    }

    public interface View extends FragmentContract.View {
        void proceedToNextScreen();

        void launchWorker();

        void showError(Throwable error);
    }

    public interface Host extends FragmentContract.Host {
        void proceedToPokemonsScreen();

        void showError(Throwable error);
    }
}

File: app/src/main/java/com/mdgd/pokemon/ui/splash/SplashContract.kt
package com.mdgd.pokemon.ui.splash

import com.mdgd.mvi.fragments.FragmentContract

class SplashContract {
    companion object {
        const val SPLASH_DELAY = 1000L
    }


    interface ViewModel : FragmentContract.ViewModel<View>

    interface View : FragmentContract.View {
        fun proceedToNextScreen()
        fun launchWorker()
        fun showError(error: Throwable?)
    }

    interface Host : FragmentContract.Host {
        fun proceedToPokemonsScreen()
    }
}
